File: 1396-count-servers-that-communicate/1396-count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        row = {}
        col = {}

        for i in range(len(grid)):
            count = 0
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    count += 1
            row[i] = count

        for i in range(len(grid[0])):
            count = 0
            for j in range(len(grid)):
                if grid[j][i] == 1:
                    count += 1
            col[i] = count
        
        ans = 0

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1 and (row[i]>1 or col[j]>1):
                    ans += 1
        
        return ans

File: 1396-count-servers-that-communicate/1396-count-servers-that-communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        int rows = grid.size(), cols = grid[0].size(), ans = 0;
        vector<int> rowData(rows,0), colData(cols,0);
        
        for(int i=0;i<rows;i++){
            for(int j=0;j<cols;j++){
                if(grid[i][j] == 1){
                    rowData[i]++;
                    colData[j]++;
                }
            }
        }

        for(int i=0;i<rows;i++){
            for(int j=0;j<cols;j++){
                if(grid[i][j] == 1){
                    if(rowData[i] > 1 || colData[j] > 1){
                        ans++;
                    }
                }
            }
        }
        return ans;
    }
};